benchmarks/perf-tool/add-filters-to-dataset.py

```diff
@@ -0,0 +1,172 @@
+import getopt
+import os
+import random
+import sys
+
+import h5py
+
+from osb.extensions.data_set import Context, HDF5DataSet
+
+"""
+Script builds complex dataset with additional attributes from exiting dataset that has only vectors. 
+Additional attributes are predefined in the script: color, taste, age. Only HDF5 format of vector dataset is supported.
+
+Script generates additional dataset of neighbours (ground truth) for each filter type. 
+
+Example of usage:
+    
+    create new hdf5 file with attribute dataset
+    add-filters-to-dataset.py ~/dev/opensearch/k-NN/benchmarks/perf-tool/dataset/data.hdf5 ~/dev/opensearch/datasets/data-with-attr True False
+    
+    create new hdf5 file with filter datasets
+    add-filters-to-dataset.py ~/dev/opensearch/k-NN/benchmarks/perf-tool/dataset/data-with-attr.hdf5 ~/dev/opensearch/datasets/data-with-filters False True
+"""
+
+class Dataset():
+    DEFAULT_INDEX_NAME = "test-index"
+    DEFAULT_FIELD_NAME = "test-field"
+    DEFAULT_CONTEXT = Context.INDEX
+    DEFAULT_TYPE = HDF5DataSet.FORMAT_NAME
+    DEFAULT_NUM_VECTORS = 10
+    DEFAULT_DIMENSION = 10
+    DEFAULT_RANDOM_STRING_LENGTH = 8
+
```

Comment:
Can we add python javadocs method equivalent to explain params?

Reply:
ack